lib/regex.ts

```diff
@@ -1,9 +1,21 @@
-const DOMAIN = '([^/?#]*)';
-const ADDRESS = '0x[a-zA-Z0-9]{40}';
-const URI = '(([^:/?#]+):)?(//([^/?#]*))?([^?#]*)(\\?([^#]*))?(#(.*))?';
-const DATETIME =
-	'([0-9]+)-(0[1-9]|1[012])-(0[1-9]|[12][0-9]|3[01])[Tt]([01][0-9]|2[0-3]):([0-5][0-9]):([0-5][0-9]|60)(\\.[0-9]+)?(([Zz])|([\\+|\\-]([01][0-9]|2[0-3]):[0-5][0-9]))';
-const REQUESTID = "[-._~!$&'()*+,;=:@%a-zA-Z0-9]*";
+const DOMAIN =
+	'^(?<domain>([^?#]*)) wants you to sign in with your Ethereum account:\\n';
+const ADDRESS = '(?<address>0x[a-zA-Z0-9]{40})\\n\\n';
+const STATEMENT = '((?<statement>[^\\n]+)\\n)?\\n';
+const URI = '(([^:?#]+):)?(([^?#]*))?([^?#]*)(\\?([^#]*))?(#(.*))';
+const URI_LINE = `URI: (?<uri>${URI}?)\\n`;
```

Comment:
Maybe, to match the ABNF grammar, the line returns should be moved to the beginning of the following item